CARDS-1355: Review PROMs questionnaires

marta- opened the pull request:
[CARDS-1355](https://phenotips.atlassian.net/browse/CARDS-1355): _Review PROMs questionnaires_

#### EQ5D 
- Updated the displayed title
- Instructions for filling out the form no longer displayed in view mode (turned section label into an info card)
- Fixed the option values (fewer difficulties should have lower score)
- Updated the score formula

#### GAD7
- Added "-7" to the displayed title, for consistency with PHQ9

#### SC
- Made all boolean questions compact
- Improved label ([CARDS-1579](https://phenotips.atlassian.net/browse/CARDS-1579): _Better name for "Check all that apply" field_)
- Reworded info card to be less cancer-specific
- Fixed wrongly formatted default value

marta- committed:
CARDS-1355: Review PROMs questionnaires - EQ5D
- Updated the displayed title
- Instructions for filling out the form no longer displayed in view mode (turned section label into an info card)
- Fixed the option values (fewer difficulties should have lower score)
- Updated the score formula

marta- committed:
CARDS-1355: Review PROMs questionnaires - GAD7
Added "-7" to the displayed title, for consistency with PHQ9

marta- committed:
CARDS-1355: Review PROMs questionnaires - SC
* Made all boolean questions compact
* Improved label (CARDS-1579: Better name for "Check all that apply" field)
* Reworded info card to be less cancer-specific

marta- committed:
CARDS-1355: Review PROMs questionnaires - SC
Fixed wrongly formatted default value

marta- committed:
CARDS-1570: Remove text 'You have selected "none"...' for non-smokers in Smoking Cessation survey form
Workaround/Quick fix:
* Rename the variable conditioning the display of the section, so that it can be referenced in javascript
* Update the formula to return null if that variable is 0 (and the section should not have been shoown anyway)

sashaandjic commented:
Not sure about these:
![image](https://user-images.githubusercontent.com/23243244/150822326-6ffca137-9080-4452-93bf-1f9c5d80d41a.png)
![image](https://user-images.githubusercontent.com/23243244/150822461-b312abf7-abd5-4713-ae9b-8f03242b86c1.png)
![image](https://user-images.githubusercontent.com/23243244/150822363-43fa99db-3a0a-448e-9496-8aba0b52dc20.png)

sdumitriu commented on proms-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/EQ5D.xml:
```diff
@@ -55,7 +55,7 @@
 			<property>
 				<name>expression</name>
-				<value>return (10 * (+@{eq5d_1:-0} + +@{eq5d_2:-0} + +@{eq5d_3:-0} + +@{eq5d_4:-0} + +@{eq5d_5:-0}) + (100 - +@{eq5d_6:-0}))</value>
+				<value>return (+@{eq5d_1:-0} + +@{eq5d_2:-0} + +@{eq5d_3:-0} + +@{eq5d_4:-0} + +@{eq5d_5:-0})</value>
```
6 is no longer used in the score?

marta- replied:
No, the score only depends on the questions with options.

sdumitriu commented on proms-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/EQ5D.xml:
```diff
@@ -22,7 +22,7 @@
 	<primaryNodeType>cards:Questionnaire</primaryNodeType>
 	<property>
 		<name>title</name>
-		<value>EQ-5D</value>
```
Should the code be put as description? Either way, the current description is not useful anymore and could be removed or expanded.

marta- replied:
> Should the code be put as description? Either way, the current description is not useful anymore and could be removed or expanded.

Addressed in 17758a0

sdumitriu commented on proms-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/GAD7.xml:
```diff
@@ -22,7 +22,7 @@
 	<property>
 		<name>title</name>
-		<value>General Anxiety Disorder</value>
+		<value>General Anxiety Disorder-7</value>
```
Isn't this ugly?

marta- replied:
Ugly, but considered informative.

sdumitriu commented on proms-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/SC.xml:
```diff
@@ -406,9 +406,9 @@
 			<property>
 				<name>text</name>
-				<value>## Quitting is an important part of cancer treatment.
+				<value>## Quitting is an important part of heath care.
```
```suggestion
				<value>## Quitting is an important part of health care.
```

sdumitriu reviewed (approved):
The titles are not consistent, a mixture of acronyms and descriptions. Should we make them all similar, either title=long, description=acronym or the other way around?

marta- committed:
CARDS-1355: Review PROMs questionnaires - SC

Fixed leftover formatting fragment.

marta- committed:
CARDS-1355: Review PROMs questionnaires - SC

Typo fix

Co-authored-by: Sergiu Dumitriu <sdumitriu@users.noreply.github.com>

marta- committed:
CARDS-1355: Review PROMs questionnaires
Improved title and description consistency: Long names as questionnaire title, short name/code as its description

marta- commented:
> Not sure about these: 

- Image 1 - Waiting for several rewording instructions across SC, this one included.
- Image 2 - Addressed in d960230 .
- Image 3 - That's the short name of the questionnaire. It's meaningful to clinicians. That card will only be seen by clinicians.

sashaandjic commented:
@marta- With the current implementation patient do see it.
![image](https://user-images.githubusercontent.com/23243244/150890262-7960f774-eef6-4709-b715-33cd860ea937.png)

sashaandjic commented:
Everything else look perfect. So if the above complaint isn't valid, I can mark this PR as `tested`.

marta- commented:
> @marta- With the current implementation patient do see Pitfalls while filling in GAD-7 survey.

Are you logging in with a patient token or with admin before going through the identification screen? Until CARDS-1550 is ready, we won't see exactly what the patient sees if we don't manually authenticate with a token.

sashaandjic commented:
> Are you logging in with a patient token or with admin before going through the identification screen? Until CARDS-1550 is ready, we won't see exactly what the patient sees if we don't manually authenticate with a token.

Ah I see. Let me check.

marta- commented:
> > @marta- With the current implementation patient do see Pitfalls while filling in GAD-7 survey.
> 
> Are you logging in with a patient token or with admin before going through the identification screen? Until CARDS-1550 is ready, we won't see exactly what the patient sees if we don't manually authenticate with a token.

Even if it doesn't work properly now, it's a permissions issue rather than a questionnaire structure issue, and should be tackled separately.

sashaandjic commented:
When I log with patient token, everything looks great ( no Pitfals)!
![image](https://user-images.githubusercontent.com/23243244/150893385-73dce864-47d6-48be-8175-fe62be6ffde0.png)

sdumitriu committed:
Merge pull request #865 from data-team-uhn/CARDS-1570

CARDS-1570: Remove text 'You have selected "none"...' for non-smokers in Smoking Cessation survey form
